frontend/src/metabase-lib/Question.ts

```diff
@@ -1129,6 +1132,14 @@ class QuestionInner {
     }
   }
 
+  setParameter(newParameter) {
+    const newParameters = this.parameters().map(oldParameter =>
+      oldParameter.id === newParameter.id ? newParameter : oldParameter,
```

Comment:
I haven't seen many examples of set functions that use their parameter's property to update the designated item in a list, or are we already be using this pattern in our app?

Normally I'd expect set functions to have a signature somewhere along the line of `setSomething(thingToSet, whereToSet, valueToSet)` or `thingToSet.setSomething(whereToSet, valueToSet)` e.g. `question.setParameter(parameterId, parameter)`

This should be clearer and more similar to other set functions. What do you think?